pkg/entity/register/worker_test.go

```diff
@@ -385,3 +401,167 @@ func TestWorker_registerEntitiesWithRetry_Success(t *testing.T) {
+	defer cancel()
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-reqsRegisteredQueue:
+				// empty the registered queue to process new requests.
+			case <-time.After(5 * time.Second):
+				cancel()
+				t.Error("Timeout while executing the test")
+			}
+		}
+	}()
+
+	hook := new(test.Hook)
+	log.AddHook(hook)
+	log.SetOutput(ioutil.Discard)
+
+	batchSizeBytes := 10000
+	w.send(ctx, batch, &batchSizeBytes)
+
+	found := false
+	var actualErrors []string
+	for _, expectedError := range expectedErrs {
+		for i, entry := range hook.AllEntries() {
+			if entry.Level != log.ErrorLevel {
+				continue
+			}
+			if val, ok := hook.AllEntries()[i].Data["error"]; ok {
+				errStr := val.(error).Error()
+				if errStr == expectedError {
+					found = true
+				}
+				actualErrors = append(actualErrors, errStr)
+			}
+		}
+	}
+	assert.True(t, found, "expected to find error message: %s \n actual errors: %s",
+		expectedErrs, actualErrors)
+
+	var actualWarnings []string
+	found = false
+	for _, expectedWarn := range expectedWarnings {
+		for i, entry := range hook.AllEntries() {
+			if entry.Level != log.WarnLevel {
+				continue
+			}
+			if val, ok := hook.AllEntries()[i].Data["error"]; ok {
+				warnStr := val.(error).Error()
+				if warnStr == expectedWarn {
+					found = true
+				}
+				actualWarnings = append(actualWarnings, warnStr)
+			}
+		}
+	}
+	assert.True(t, found, "expected to find warn message: %s \n actual warns: %s",
+		expectedWarnings, actualWarnings)
+}
+
+func TestWorker_send_Logging_VerboseDisabled(t *testing.T) {
+	// When the request is successful but some entities fail, we log only when verbose is enabled.
+	reqsToRegisterQueue := make(chan fwrequest.EntityFwRequest, 0)
+	reqsRegisteredQueue := make(chan fwrequest.EntityFwRequest, 0)
+
+	agentIdentity := func() entity.Identity {
+		return entity.Identity{ID: 12}
+	}
+
+	client := &fakeClient{
+		ids: []entity.ID{13, 14},
+		// no err from backend.
+		err: nil,
+	}
+
+	config := WorkerConfig{
+		VerboseLogLevel: 0,
+	}
+	w := NewWorker(agentIdentity, client, backoff.NewDefaultBackoff(), reqsToRegisterQueue, reqsRegisteredQueue, config)
+
+	batch := map[entity.Key]fwrequest.EntityFwRequest{
+		entity.Key("error"): {
+			Data: protocol.Dataset{
+				Entity: entity.Fields{Name: "error"},
+			},
+		},
+		entity.Key("warnings"): {
+			Data: protocol.Dataset{
+				Entity: entity.Fields{Name: "warnings"},
+			},
+		},
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-reqsRegisteredQueue:
+				// empty the registered queue to process new requests.
+			case <-time.After(5 * time.Second):
+				cancel()
+				t.Error("Timeout while executing the test")
+			}
+		}
+	}()
+
+	hook := new(test.Hook)
+	log.AddHook(hook)
+	log.SetOutput(ioutil.Discard)
+
+	batchSizeBytes := 10000
+	w.send(ctx, batch, &batchSizeBytes)
+
+	assert.Empty(t, hook.Entries)
```

Comment:
This bit of code will panic and end in race conditions (flaky tests). I suggest looking at how it was done in [here](https://github.com/newrelic/infrastructure-agent/blob/master/pkg/integrations/v4/dm/emitter_test.go#L255) to make it more solid.